enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,256 @@
+    $ oc delete -n kube-system deployment.apps/csi-snapshot-controller deployment.apps/csi-snapshot-webhook   
+    $ oc delete -n openshift-storage daemonset.apps/topolvm-node
+    $ oc delete -n openshift-storage deployment.apps/topolvm-controller
+    $ oc delete -n openshift-storage configmaps/lvmd
+    $ oc delete oc get storageclasses.storage.k8s.io/topolvm-provisioner
+    ```
+7. The cluster processes the deletions and the process is complete
+
+### API Extensions
+
+- None
+
+### Topology Considerations
+
+#### Single-node Deployments or MicroShift
+
+The changes proposed here only affect MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+- **Rebase Changes:** 
+  - LVMS and the snapshotting manifests will be moved from `microshift/assets/components/lvms` and 
+  `microshift/assets/components/csi-snapshot-controller` to `microshift/assets/optional/lvms` and 
+  `microshift/assets/optional/csi-snapshot-controller`. These paths will be updated in the 
+    `microshift/scripts/auto-rebase/lvms_assets.yaml` to reflect the new paths.
+  - LVMS image digests will be moved from `microshift/assets/release/release-$ARCH.json` to 
+    `microshift/assets/optional/lvms/release-$ARCH.json`.
+  - CSI Snapshotter image digests will be moved from `microshift/assets/release/release-$ARCH.json` to 
+      `microshift/assets/optional/csi-snapshot-controller/release-$ARCH.json`.
+  - During rebasing, the image digests for LVMS and CSI Snapshotter will be parsed from the LVMS bundle (like they 
+    are now) and the digests will be written to their respective `release-$ARCH.json` files.
+
+- **Mainline Code Changes:**
+  - MicroShift source code will no longer manage LVMS or CSI manifest deployment or its configuration. The 
+    microshift service manager which handles these components will be removed entirely. The following files will be 
+    deleted:
+    - `microshift/pkg/components/csi-snapshot-controller.go`
+    - `microshift/pkg/components/storage.go`
+    - `microshift/pkg/components/render_test.go`, which only tests LVMS parameter rendering
+    - `microshift/pkg/assets/storage.go`
+  - Additionally, certain functions related to managing LVMS will be deleted:
+    - `microshift/pkg/components/render_test.go:startCSIPlugin()`
+    - `microshift/pkg/components/render_test.go:startCSISnapshotterController()`
+  - MicroShift also contains logic to determine if LVM is installed and attempt a best-guess at which volume group, 
+    if any, should be set in the LVMD config. This code will need to be removed and implemented as:
+    - Specifying LVM as an RPM dependency of `microshift-lvms`.
+    - A `microshift-lvms` rpm post-install script that determines what volume group, if any, should be specified.
+
+- **RPM Spec**
+  - New RPM specs will be written for `microshift-lvms` and `microshift-lvms-snapshotting` packages. These have the 
+    following characteristics:
+    - `microshift-lvms-snapshotting` will depend on `microshift-lvms`.
```

Comment:
I think we could drop that dependency and let LVMS do dynamic adjustment to wether VolumeSnapshotClass Support exists in the cluster.

Reply:
That's a good idea, but it also means this work would be blocked until the LVMS change was accepted and implemented, which doesn't make sense.  Instead, let this change and that one be developed in parallel at their own velocities. Whenever the LVMS change is implemented, it'd just require a one-line change in the microshift rpm spec.

Reply:
Pro: Unblocks engineering on both fronts
Contra: this will mean that the rpm in the future might need to keep that dependency. Maybe Im just wondering why we dont have a microshift-snapshotting rpm with all the snapshotting stuff?